src/accelerate/utils/modeling.py

```diff
@@ -350,17 +352,34 @@ def set_module_tensor_to_device(
             elif param_cls.__name__ in ["QTensor", "QBitsTensor"]:
                 new_value = torch.nn.Parameter(new_value, requires_grad=old_value.requires_grad).to(device)
             elif param_cls.__name__ in ["AffineQuantizedTensor"]:
-                new_value = torch.nn.Parameter(
-                    param_cls(
-                        new_value.layout_tensor,
-                        new_value.block_size,
-                        new_value.shape,
-                        new_value.quant_min,
-                        new_value.quant_max,
-                        new_value.zero_point_domain,
-                    ),
-                    requires_grad=old_value.requires_grad,
-                ).to(device)
+                if (
+                    importlib.util.find_spec("torchao") is not None
+                    and importlib.metadata.version("torchao") >= "0.7.0"
+                ):
+                    # TorchAO v0.7.0 made layout_tensor an internal private variable and exposed tensor_impl
+                    new_value = torch.nn.Parameter(
+                        param_cls(
+                            new_value.tensor_impl,
+                            new_value.block_size,
+                            new_value.shape,
+                            new_value.quant_min,
+                            new_value.quant_max,
+                            new_value.zero_point_domain,
+                        ),
+                        requires_grad=old_value.requires_grad,
+                    ).to(device)
+                else:
+                    new_value = torch.nn.Parameter(
+                        param_cls(
+                            new_value.layout_tensor,
+                            new_value.block_size,
+                            new_value.shape,
+                            new_value.quant_min,
+                            new_value.quant_max,
+                            new_value.zero_point_domain,
+                        ),
+                        requires_grad=old_value.requires_grad,
+                    ).to(device)
```

Comment:
Only the name of the first argument changed, right? Could the code be simplified to something like:

```python
if ...:
    args = (new_value.tensor_impl,)
else:
    args = (new_value.layout_tensor,)
args = args + (new_value.block_size, ...)
new_value = torch.nn.Parameter(param_cls(*args)).to(device)